daliuge-translator/dlg/dropmake/dm_utils.py

```diff
@@ -616,11 +734,23 @@ def convert_construct(lgo):
                     if cond1 and (k_old in old_newnew_gather_map):
                         link["from"] = old_newnew_gather_map[k_old]
                     # print("from %d to %d to %d" % (link['from'], k_old, link['to']))
-
-    # print('%d nodes in lg after construct conversion' % len(lgo['nodeDataArray']))
+        if old_new_subgraph_map:
+            for key, value in old_new_subgraph_map.items():
+                link = {}
+                link["from"] = key
+                link["to"] = value
+                lgo['linkDataArray'].append(link)
+    # print()
+    # for link in lgo['linkDataArray']:
+    #     print(keyNameMap[link['from']], ' to ', keyNameMap[link['to']])
+    print('%d nodes in lg after construct conversion' % len(lgo['nodeDataArray']))
```

Comment:
The `test_tool_trans.py` test failures are due to these `print` statements. I will remove in the final implementation.